cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/constants/AnomalyDetectorConfig.java

```diff
@@ -189,6 +189,24 @@
       + "detector will run to detect disk failures. If this interval time is not specified, disk failure detector "
       + "will run with interval specified in " + ANOMALY_DETECTION_INTERVAL_MS_CONFIG + ".";
 
+  /**
+   * <code>kafka.broker.failure.detection.enable</code>
+   */
+  public static final String KAFKA_BROKER_FAILURE_DETECTION_ENABLE_CONFIG = "kafka.broker.failure.detection.enable";
+  public static final Boolean DEFAULT_KAFKA_BROKER_FAILURE_DETECTION_ENABLE = false;
+  public static final String KAFKA_BROKER_FAILURE_DETECTION_ENABLE_DOC = "Whether to use the Kafka API to detect broker failures "
+      + "instead of ZooKeeper.";
+
+  /**
+   * <code>broker.failure.detection.interval.ms</code>
+   */
+  public static final String BROKER_FAILURE_DETECTION_INTERVAL_MS_CONFIG = "broker.failure.detection.interval.ms";
+  public static final Long DEFAULT_BROKER_FAILURE_DETECTION_INTERVAL_MS = null;
+  public static final String BROKER_FAILURE_DETECTION_INTERVAL_MS_DOC = "The interval in millisecond that broker failure "
+          + "detector will run to detect broker failures. If this interval time is not specified, disk failure detector "
+          + "will run with interval specified in " + ANOMALY_DETECTION_INTERVAL_MS_CONFIG + ". This is only used when "
+          + KAFKA_BROKER_FAILURE_DETECTION_ENABLE_CONFIG + " is set to 'true'.";
+
```

Comment:
1. How do we handle the use of (1) `broker.failure.alert.threshold.ms` and (2) `broker.failure.self.healing.threshold.ms` with adminClient-based broker failure detector?
2. Do we need to update the JavaDoc of `SelfHealingNotifier`?